libraries/libfc/include/fc/crypto/public_key.hpp

```diff
@@ -28,7 +30,8 @@ namespace fc { namespace crypto {
          public_key( const public_key& ) = default;
          public_key& operator= (const public_key& ) = default;
 
-         public_key( const signature& c, const sha256& digest, bool check_canonical = true );
+         public_key( const signature& c, const sha256& digest );
+         public_key( const signature& c, const sha256& digest, bool check_r1_canonical ); // potential canonical check only for r1 sigs
```

Comment:
Seems like this should get the `enum` treatment like the other changes.

Reply:
I thought it was used in just one location, and where it is used we are passing an expression (`!is_builtin_activated(builtin_protocol_feature_t::allow_non_canonical_signatures)`), not passing a blind constant like true/false), so I think bool is fine here even if it seems admitedly inconsistent.

Reply:
That would makke the call location change from:
```
      auto recovered = fc::crypto::public_key(
         s, *digest, !context.control.is_builtin_activated(builtin_protocol_feature_t::allow_non_canonical_signatures));
```

to

```
      auto recovered = fc::crypto::public_key(
         s, *digest,
         context.control.is_builtin_activated(builtin_protocol_feature_t::allow_non_canonical_signatures)
            ? fc::check_canonical_t::no
            : fc::check_canonical_t::yes);
```

which doesn't seem better to me.